File: src/main/kotlin/utils/Waiter.java
package utils;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.WaitForSelectorState;

public final class Waiter {
    private Waiter() {}

    public static void visible(Page page, String selector) {
        page.waitForSelector(selector, new Page.WaitForSelectorOptions()
                .setState(WaitForSelectorState.VISIBLE));
    }

    public static void hidden(Page page, String selector) {
        page.waitForSelector(selector, new Page.WaitForSelectorOptions()
                .setState(WaitForSelectorState.HIDDEN));
    }
}

File: src/main/kotlin/utils/Waiter.kt
package utils

import com.microsoft.playwright.Page
import com.microsoft.playwright.options.WaitForSelectorState

object Waiter {
    fun waitVisible(page: Page, selector: String) {
        page.waitForSelector(selector, Page.WaitForSelectorOptions()
            .setState(WaitForSelectorState.VISIBLE))
    }

    fun waitHidden(page: Page, selector: String) {
        page.waitForSelector(selector, Page.WaitForSelectorOptions()
            .setState(WaitForSelectorState.HIDDEN))
    }

    fun waitUrl(page: Page, pattern: String) {
        page.waitForURL(pattern)
    }
}
